libs/checkpoint/langgraph/checkpoint/serde/base.py

```diff
@@ -10,8 +10,8 @@ class SerializerProtocol(Protocol):
     Valid implementations include the `pickle`, `json` and `orjson` modules.
     """
 
-    def dumps(self, obj: Any) -> bytes:
+    def dumps(self, obj: Any) -> tuple[str, bytes]:
```

Comment:
if we want to minimize breaking changes we could do this change in a fully backwards compatible way I think, by making these new methods, and any old serializer with only old methods that is passed in we wrap it to implement the new methods delegating to old ones